tensorflow/compiler/xla/literal_util.cc

```diff
@@ -917,6 +933,16 @@ LiteralUtil::GetMutableArraySlice(Literal* literal) {
 }
 
 template <>
+/* static */ tensorflow::gtl::MutableArraySlice<half>
```

Comment:
It occurs to me that there is a latent endian-ness bug here. If you were to use this code on a big-endian architecture, the layout of the encoded bytes would not match the byte layout in memory.

Not sure what to do here. I suspect for now, leave a // TODO: add big-endian support and move on.

I suspect to fix this right, we should add logic to perform byteswapping on serialization or deserialization of a half literal.